packages/netsuite-adapter/src/client/client.ts

```diff
@@ -423,14 +430,24 @@ export default class NetsuiteClient {
     }, executor)
   }
 
-  private async listFilePaths(executor: CommandActionExecutorType): Promise<string[]> {
-    const TOP_LEVEL_FOLDER_NAMES = [`${SDF_PATH_SEPARATOR}${SUITE_SCRIPTS_FOLDER_NAME}`,
-      `${SDF_PATH_SEPARATOR}${TEMPLATES_FOLDER_NAME}`, `${SDF_PATH_SEPARATOR}${WEB_SITE_HOSTING_FILES_FOLDER_NAME}`]
-    const operationResults = _.flatten(await Promise.all(
-      TOP_LEVEL_FOLDER_NAMES.map(async folderName =>
-        this.executeProjectAction(COMMANDS.LIST_FILES, { folder: folderName }, executor))
-    ))
-    return _.flatten(operationResults.map(operationResult => makeArray(operationResult.data)))
+  private async listFilePaths(executor: CommandActionExecutorType, filePathRegexSkipList: RegExp[]):
+    Promise<{ listedPaths: string[]; failedPaths: string[] }> {
+    const failedPaths: string[] = []
+    const operationResults = (await Promise.all(
+      fileCabinetTopLevelFolders
+        .filter(folderName => filePathRegexSkipList.every(regex => !regex.test(folderName)))
+        .map(async folderName =>
+          this.executeProjectAction(COMMANDS.LIST_FILES, { folder: folderName }, executor)
+            .catch(() => {
+              log.debug(`Adding ${folderName} path to skip list`)
+              failedPaths.push(folderName)
+              return undefined
+            }))
+    )).filter(values.isDefined)
```

Comment:
nit: might be a bit cleaner without side effects
```suggestion
    const operationResults = await Promise.all(
      fileCabinetTopLevelFolders
        .filter(folderName => filePathRegexSkipList.every(regex => !regex.test(folderName)))
        .map(async folder => [
          folder,
          await this.executeProjectAction(COMMANDS.LIST_FILES, { folder }, executor).catch(() => undefined),
        ])
    )
    const [failedFolders, successfulFolders] = _.partition(operationResults, ([_name, result]) => result === undefined)
    ...
```

Reply:
Decided to stick with my implementation as I find it clearer although the side effect